src/lib/DataTable/utils.ts

```
@@ -0,0 +1,273 @@
+const matchesSearch = <T extends DataRow>(
+  row: T,
+  query: string,
+  fields: (keyof T & string)[]
+): boolean => {
+  const term = query.trim().toLowerCase();
+  if (term === '') {
+    return true;
+  }
+  return fields.some((field) => {
+    const value = row[field];
+    return value !== null && value !== undefined && String(value).toLowerCase().includes(term);
```

ESLint

[error] undefined is not allowed. Use null or proper type checking instead.

(no-restricted-syntax)